modules/project-factory/variables.tf

```diff
@@ -260,6 +273,18 @@ variable "factories_data" {
               description = optional(string)
             }))
           })), {})
+          outputs_bucket = optional(object({
```

Comment:
I would change this to be either a bucket name, or bucket details for a new bucket. And add validation to ensure only one of the two is passed.